frontend/src/metabase/query_builder/components/AlertListPopoverContent/AlertListItem.tsx

```diff
@@ -0,0 +1,149 @@
+
+import { AlertCreatorTitle } from "./AlertCreatorTitle";
+import { AlertScheduleText } from "./AlertScheduleText";
+import { UnsubscribedListItem } from "./UnsubscribedListItem";
+
+type AlertListItemProps = {
+  alert: Alert;
+  highlight: boolean;
+  setMenuFreeze: (freeze: boolean) => void;
+  closeMenu: () => void;
+  onUnsubscribe: (alert: Alert) => void;
+};
+
+export const AlertListItem = ({
+  alert,
+  highlight,
+  setMenuFreeze,
+  closeMenu,
+  onUnsubscribe,
+}: AlertListItemProps) => {
+  const user = useSelector(getUser);
+
+  const dispatch = useDispatch();
+
+  const [unsubscribingProgress, setUnsubscribingProgress] = useState<
+    string | null
+  >(null);
+  const [hasJustUnsubscribed, setHasJustUnsubscribed] = useState(false);
+  const [editing, setEditing] = useState(false);
+
+  const handleUnsubscribe = async () => {
+    try {
+      setUnsubscribingProgress(t`Unsubscribing...`);
+      await dispatch(unsubscribeFromAlert(alert));
+      setHasJustUnsubscribed(true);
+      onUnsubscribe(alert);
+    } catch (e) {
+      setUnsubscribingProgress(t`Failed to unsubscribe`);
+    }
+  };
+
+  const onEdit = () => {
+    setMenuFreeze(true);
+    setEditing(true);
+  };
+
+  const onEndEditing = (shouldCloseMenu = false) => {
+    setMenuFreeze(false);
+    setEditing(false);
+    if (shouldCloseMenu) {
+      closeMenu();
+    }
+  };
+
+  const isAdmin = user && user.is_superuser;
+  const isCurrentUser = user && alert.creator.id === user.id;
+
+  const emailChannel = alert.channels.find(c => c.channel_type === "email");
+  const emailEnabled = emailChannel && emailChannel.enabled;
+  const slackChannel = alert.channels.find(c => c.channel_type === "slack");
+  const slackEnabled = slackChannel && slackChannel.enabled;
+
+  if (hasJustUnsubscribed) {
+    return <UnsubscribedListItem />;
+  }
+
+  return (
+    <li
+      className={cx(CS.flex, CS.p3, CS.textMedium, CS.borderBottom, {
+        [CS.bgLightBlue]: highlight,
+      })}
+    >
+      <Icon name="alert" size="20" />
+      <div className={cx(CS.full, CS.ml2)}>
+        <div className={cx(CS.flex, "align-top")}>
+          <div>{user && <AlertCreatorTitle alert={alert} user={user} />}</div>
+          <div
+            className={cx(CS.mlAuto, CS.textBold, CS.textSmall)}
+            style={{
+              transform: `translateY(4px)`,
+            }}
+          >
+            {(isAdmin || isCurrentUser) && (
+              <a className={CS.link} onClick={onEdit}>{jt`Edit`}</a>
+            )}
+            {!isAdmin && !unsubscribingProgress && (
+              <a
+                className={cx(CS.link, CS.ml2)}
+                onClick={handleUnsubscribe}
+              >{jt`Unsubscribe`}</a>
+            )}
+            {!isAdmin && unsubscribingProgress && (
+              <span> {unsubscribingProgress}</span>
+            )}
+          </div>
+        </div>
+
+        {
+          // To-do: @kdoh wants to look into overall alignment
```

Comment:
wat? let's remove it

Reply:
🤣  aww cmon, these are my favorite kinds of comments